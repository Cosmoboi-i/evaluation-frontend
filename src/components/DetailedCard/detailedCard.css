.detail-card {
  width: 620px;
  height: 800px;
  border: 1px solid black;
  border-radius: 10px;

  display: flex;
  flex-direction: column
}

.detail-card img {
  width: 620px;
  height: 330px;
  border-radius: 10px 10px 0px 0px;
  margin-bottom: 1px;
}

.detail-card .card-content {
  background: black;
  flex-grow: 1;
  border-radius: 0px 0px 10px 10px;
  color: white;
  padding: 0 1.8rem;

  display: flex;
  flex-direction: column;
  align-items: center;
}

.detail-card .card-heading {
  padding: 1.3rem;
  font-size: 1.8rem;
  font-family: "big shoulders text";
}

.detail-card .card-content {
  font-size: 1.2rem;
  text-align: center;

  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

.detail-card .card-desc {
  display: -webkit-box;
  -webkit-box-orient: vertical;
  overflow: hidden;
  -webkit-line-clamp: 10;
}

.detail-card .venue {
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 2;
  overflow: hidden;

}

.detail-card .reactions {
  padding: 0.5rem 0
}

.detail-card .reactions {
  width: 100%;
  display: flex;
  justify-content: space-between;
}

.detail-card .register {
  color: rgb(20, 202, 90);

  display: flex;
  align-items: center;
}

.detail-card .registered {
  padding-left: 5px;
}

.detail-card .bookmark {
  color: rgb(250, 82, 82);
}

.detail-card .register-button {
  width: 100%;
  height: 4rem;
}

.detail-card .no-seats{
  color: rgb(252, 246, 50);

  display: flex;
  align-items: center;
}

.detail-card .register-button > button {
  background: white;
  color: black;
  border: none;
  padding: 0.5rem 1.5rem;
  border-radius: 0.5rem;

  font-size: 1.2rem;
  font-weight: 500;
}