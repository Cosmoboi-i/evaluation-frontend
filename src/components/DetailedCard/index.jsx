import React, { useEffect } from "react";
import "./detailedCard.css";
import Proptypes from "prop-types";

export default function DetailedCard(props) {
  useEffect(() => {
    console.log("props", props);
  });
  return (
    <div className="detail-card">
      <img src={props.imgUrl}></img>
      <div className="card-content">
        <div className="card-heading">{props.name}</div>
        <div className="card-content">
          <div className="card-desc">{props.description}</div>
          <div>
            <div className="venue">
              <b>VENUE:</b> {props.venue}
            </div>
            <div className="venue">
              <b>DATE:</b> {props.datetime}
            </div>
          </div>
        </div>
        <div className="reactions">
          {props.areSeatsAvailable ? (
            <div className="register">
              <i
                className={`${
                  props.isRegistered ? "fa-solid" : "fa-regular"
                } fa-circle-check fa-2x`}
              ></i>
              <div className="registered">
                {props.isRegistered ? "REGISTERED" : "REGISTER"}
              </div>
            </div>
          ) : (
            <div className="register no-seats">
              <i className="fa-solid fa-circle-xmark fa-2x"></i>
              <div className="registered">NO SEATS AVAILABLE</div>
            </div>
          )}
          <i
            className={`${
              props.isBookmarked ? "fa-solid" : "fa-regular"
            } fa-bookmark fa-2x bookmark`}
            onClick={() =>
              props.updateEvent(props.id, {
                isBookmarked: !props.isBookmarked,
              })
            }
          ></i>
        </div>
        <div className="register-button">
          {props.areSeatsAvailable ? (
            <button
              onClick={() =>
                props.updateEvent(props.id, {
                  isRegistered: !props.isRegistered,
                })
              }
            >
              {props.isRegistered ? "UNREGISTER" : "REGISTER"}
            </button>
          ) : (
            <></>
          )}
        </div>
      </div>
    </div>
  );
}

DetailedCard.propTypes = {
  id: Proptypes.number.isRequired,
  name: Proptypes.string.isRequired,
  description: Proptypes.string.isRequired,
  venue: Proptypes.string.isRequired,
  datetime: Proptypes.string.isRequired,
  timezone: Proptypes.string.isRequired,
  areSeatsAvailable: Proptypes.bool.isRequired,
  isRegistered: Proptypes.bool.isRequired,
  isBookmarked: Proptypes.bool.isRequired,
  imgUrl: Proptypes.string.isRequired,
  updateEvent: Proptypes.func.isRequired,
};
