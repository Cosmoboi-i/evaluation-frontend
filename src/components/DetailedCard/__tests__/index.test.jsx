import React from "react";
import { render } from "@testing-library/react";
import DetailedCard from "..";

describe("DetailedCard", () => {
  it("should render properly", () => {
    const { asFragment } = render(<DetailedCard {} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
