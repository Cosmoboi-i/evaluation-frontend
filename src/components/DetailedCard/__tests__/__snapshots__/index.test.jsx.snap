// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailedCard should render properly 1`] = `
<DocumentFragment>
  <div
    class="detail-card"
  >
    <img />
    <div
      class="card-content"
    >
      <div
        class="card-heading"
      />
      <div
        class="card-content"
      >
        <div
          class="card-desc"
        />
        <div>
          <div
            class="venue"
          >
            <b>
              VENUE:
            </b>
             
          </div>
          <div
            class="venue"
          >
            <b>
              DATE:
            </b>
             
          </div>
        </div>
      </div>
      <div
        class="reactions"
      >
        <div
          class="register no-seats"
        >
          <i
            class="fa-solid fa-circle-xmark fa-2x"
          />
          <div
            class="registered"
          >
            NO SEATS AVAILABLE
          </div>
        </div>
        <i
          class="fa-regular fa-bookmark fa-2x bookmark"
        />
      </div>
      <div
        class="register-button"
      />
    </div>
  </div>
</DocumentFragment>
`;
